src/test/java/reposense/model/RepoLocationTest.java

```diff
@@ -0,0 +1,63 @@
+package reposense.model;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import reposense.parser.InvalidLocationException;
+import reposense.util.AssertUtil;
+
+public class RepoLocationTest {
+    @Test
+    public void repoLocation_validRepoUrl_success() throws Exception {
+        // valid url without specifying branch
+        assertValidLocation("https://github.com/reposense/testrepo-Beta.git",
+                "testrepo-Beta", "reposense");
+        assertValidLocation("https://github.com/reposense/testrepo-Delta.git",
+                "testrepo-Delta", "reposense");
+    }
+
+    @Test
+    public void repoLocation_invalidRepoUrl_throwsInvalidLocationException() {
+        // ftp url should be rejected
+        assertInvalidLocation("ftp://github.com/reposense/testrepo-Delta.git");
+        // non GitHub url should rejected
+        assertInvalidLocation("https://gitlab.com/reposense/RepoSense.git");
+        // url without organisation name should be rejected
+        assertInvalidLocation("https://github.com/reposense.git");
+        // url without repo name should be rejected
+        assertInvalidLocation("https://github.com/reposense/.git");
+        assertInvalidLocation("https://github.com/reposense");
+        // url without a .git suffix should be rejected
+        assertInvalidLocation("https://github.com/reposense/RepoSensegit");
+        assertInvalidLocation("https://github.com/reposense/RepoSense-git");
+        assertInvalidLocation("https://github.com/reposense/RepoSense.gi");
+        assertInvalidLocation("https://github.com/reposense/RepoSense");
+    }
+
+    @Test
+    public void repoLocation_repoUrlWithASpecifiedBranch_throwsInvalidLocationException() {
+        // reject both repos with and without .git
+        assertInvalidLocation("https://github.com/reposense/testrepo-Beta/tree/add-config-json");
+        assertInvalidLocation("https://github.com/reposense/testrepo-Beta.git/tree/add-config-json");
+    }
+
+    @Test
+    public void repoLocationParser_parseEmptyString_noInvalidLocationException() throws Exception {
```

Comment:
Maybe `Success` is enough? @dcshzj Your idea?

Reply:
Sorry, are you referring to all the tests for invalid URLs, or just this empty string test?

Reply:
so do I have to change it so 'Success'?

Reply:
@dcshzj I only meant for this one only, `repoLocationParser_parseEmptyString_noInvalidLocationException()` -> `repoLocationParser_parseEmptyString_Success()`

Reply:
I see, yes I think it should be renamed to `success` (small s).